### PR TITLE
Brandtbowling/47 event card

### PR DESCRIPTION
**Pull request by Brandt-Bowling**
Generic card type used for events currently but could be adapted to other types (sermon cards?)

Mobile hides the description field while larger screens show it as per figma design
Screenshot 2023-08-07 at 9 21 24 PM
Screenshot 2023-08-07 at 9 21 57 PM

### CODE REVIEW

File: frontend/src/components/shared/card.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      <div className="flex flex-col gap-1 items-start w-[8.75rem] md:gap-3 md:w-[20rem]">
+        <p className="mb-0 text-[0.625rem] leading-[0.9375rem] text-black font-medium tracking-[0.0375rem] md:text-base md:leading-[1.3125rem] md:tracking-[0.0525rem]">
+          {date}
+        </p>
+        <h3 className="text-lg leading-[1.25rem] font-semibold text-black md:leading-[1.8rem] md:text-2xl">
+          {name}
+        </h3>
+        <div className="flex items-center gap-1">
+          <img className="w-4 h-4 mb-0" src={mapPinLogo} alt="Map" />
+          <p className="mb-0 text-black text-[0.625rem] leading-[0.9375rem] font-medium tracking-[0.0375rem] md:text-base md:leading[1.3125rem] md:tracking-[0.0525rem]">
```
Comment (thedomcab) [inline frontend/src/components/shared/card.js:22]:
```suggestion
          <p className="mb-0 text-black text-[0.625rem] leading-[0.9375rem] font-medium tracking-[0.0375rem] md:text-base md:leading-[1.3125rem] md:tracking-[0.0525rem]">
```

Comment (Brandt-Bowling) [reply frontend/src/components/shared/card.js:22]: I was basically going off of the developer settings of the figma design and using the matching tailwind class. So I saw `line-height: 150% /* 1.3125rem */` and used that verbatim.

File: frontend/src/components/shared/card.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import mapPinLogo from "../../images/icons/map-pin.svg";
+
+const Card = ({ type, date, name, location, description }) => {
+  return (
+    <div className="flex flex-col items-center gap-2 w-[9.75rem] h-[12.5rem] pb-[2.19rem] mx-auto rounded-xl border border-solid border-b-Neutral-900 shadow-md overflow-hidden md:w-[23.75rem] md:h-[30rem] md:pb-[3.13rem] md:gap-5">
+      <img
+        className="h-[6.15788rem] w-[10.94738rem] flex-shrink-0 mb-0 md:h-60 md:w-[26.66669rem]"
+        src={prayerGatheringImg}
+        alt="Prayer Gathering"
+      />
```
Comment (thedomcab) [inline frontend/src/components/shared/card.js:12]: I wonder should this be passed into the component, if we using Card as a template for reusability.  If not, I am okay with this. I will approve it, if we want to leave as is.  Thanks for the work. looking good.

**Review by thedomcab — APPROVED** [review]

File: frontend/src/components/shared/card.js (hunk 2 above)
Comment (Brandt-Bowling) [reply frontend/src/components/shared/card.js:12]: Yeah I was thinking about that after I finished, that it's not super reusable for other types of cards. I could either work on that as a part of this PR or maybe make it a separate task?

Comment (thedomcab) [reply frontend/src/components/shared/card.js:12]: It is up to you. I am fine with either way.  Thanks for the response.

Comment (Brandt-Bowling) [reply frontend/src/components/shared/card.js]: I tried making Card more reusable and created a specific EventCard that utilizes it. Any feedback is appreciated.

File: frontend/src/components/shared/card.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
+
+const Card = ({ img, date, title, attributes }) => {
```
Comment (thedomcab) [inline frontend/src/components/shared/card.js:3]:
```suggestion
const Card = ({ img, date, title, children }) => {
```

File: frontend/src/components/shared/card.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        alt="Background"
+      />
+      <div className="flex flex-col gap-1 items-start w-[8.75rem] md:gap-3 md:w-[20rem]">
+        <p className="mb-0 text-[0.625rem] leading-[0.9375rem] text-black font-medium tracking-[0.0375rem] md:text-base md:leading-[1.3125rem] md:tracking-[0.0525rem]">
+          {date}
+        </p>
+        <h3 className="text-lg leading-[1.25rem] font-semibold text-black md:leading-[1.8rem] md:text-2xl">
+          {title}
+        </h3>
+        {attributes}
```
Comment (thedomcab) [inline frontend/src/components/shared/card.js:18]:
```suggestion
        {children}
```

File: frontend/src/components/page-events/eventCard.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+import Card from "../shared/card";
+import prayerGatheringImg from "../../images/prayer-gathering.png";
+import mapPinLogo from "../../images/icons/map-pin.svg";
+
+const EventCard = ({ title, date, location, img, description }) => {
+  const attributes = (
+    <div>
+      <div className="flex items-center gap-1">
+        <img className="w-4 h-4 mb-0" src={mapPinLogo} alt="Map" />
```
Comment (thedomcab) [inline frontend/src/components/page-events/eventCard.js:10]: Recent changes by @graceschae  she added StaticImage  Component from Gatsby: 
```suggestion
        <StaticImage className="w-4 h-4 mb-0" src="../../images/icons/map-ping.svg" alt="Map" />
```

File: frontend/src/components/shared/card.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
+
+const Card = ({ img, date, title, attributes }) => {
+  return (
+    <div className="flex flex-col items-center gap-2 w-[9.75rem] h-[12.5rem] pb-[2.19rem] mx-auto rounded-xl border border-solid border-b-Neutral-900 shadow-md overflow-hidden md:w-[23.75rem] md:h-[30rem] md:pb-[3.13rem] md:gap-5">
+      <img
+        className="h-[6.15788rem] w-[10.94738rem] flex-shrink-0 mb-0 md:h-60 md:w-[26.66669rem]"
+        src={img}
+        alt="Background"
```
Comment (thedomcab) [inline frontend/src/components/shared/card.js:9]: Use StaticImage
```suggestion
      <StaticImage
        className="h-[6.15788rem] w-[10.94738rem] flex-shrink-0 mb-0 md:h-60 md:w-[26.66669rem]"
        src={img}
        alt="Background" />
```

File: frontend/src/pages/events/index.js [hunk 7]
```diff
@@ -4,11 +4,22 @@ import Layout from "../../components/layout";
...
+import EventCard from "../../components/page-events/eventCard";
 
 const EventsPage = () => (
   <Layout>
     <SundayCelebBarEvents />
     <div>Event list section</div>
+    <EventCard
+      title={"Prayer Gathering"}
+      date={"Th, May 25, 2023"}
+      img={PrayerGathering}
```
Comment (thedomcab) [inline frontend/src/pages/events/index.js:17]: use relative path for the use of StaticImage
```suggestion
      img={../../images/prayer-gathering.png}
```

**Comment by Brandt-Bowling** [comment]
Some of the suggestions are raising errors. I'll address them

File: frontend/src/components/page-events/eventCard.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+import Card from "../shared/card";
+import prayerGatheringImg from "../../images/prayer-gathering.png";
+import mapPinLogo from "../../images/icons/map-pin.svg";
+
+const EventCard = ({ title, date, location, img, description }) => {
+  const attributes = (
+    <div>
+      <div className="flex items-center gap-1">
+        <StaticImage className="w-4 h-4 mb-0" src="../../images/icons/map-ping.svg" alt="Map" />
```
Comment (thedomcab) [inline frontend/src/components/page-events/eventCard.js:10]:
```suggestion
        <StaticImage className="w-4 h-4 mb-0" src="../../images/icons/map-pin.svg" alt="Map" />
```

File: frontend/src/components/page-events/eventCard.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+import Card from "../shared/card";
+import prayerGatheringImg from "../../images/prayer-gathering.png";
+import mapPinLogo from "../../images/icons/map-pin.svg";
```
Comment (thedomcab) [inline frontend/src/components/page-events/eventCard.js:4]:
```suggestion
import { StaticImage } from "gatsby-plugin-image";
```

File: frontend/src/components/page-events/eventCard.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        {description}
+      </p>
+    </div>
+  );
+
+  return (
+    <Card
+      title={title}
+      date={date}
+      img={prayerGatheringImg} // TODO: make dynamic
```
Comment (thedomcab) [inline frontend/src/components/page-events/eventCard.js:25]: img
```suggestion
      img={img} // TODO: make dynamic
```

File: frontend/src/components/shared/card.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
+
+const Card = ({ img, date, title, children }) => {
+  return (
+    <div className="flex flex-col items-center gap-2 w-[9.75rem] h-[12.5rem] pb-[2.19rem] mx-auto rounded-xl border border-solid border-b-Neutral-900 shadow-md overflow-hidden md:w-[23.75rem] md:h-[30rem] md:pb-[3.13rem] md:gap-5">
+      <StaticImage
+        className="h-[6.15788rem] w-[10.94738rem] flex-shrink-0 mb-0 md:h-60 md:w-[26.66669rem]"
+        src={img}
+        alt="Background" />
+      />
```
Comment (thedomcab) [inline frontend/src/components/shared/card.js:10]:
```suggestion
   {img}
```

File: frontend/src/pages/events/index.js [hunk 12]
```diff
@@ -4,11 +4,22 @@ import Layout from "../../components/layout";
 import Seo from "../../components/seo";
 import Instagram from "../../components/instaBar";
 import SundayCelebBarEvents from "../../components/page-events/sundayCelebBarEvents";
+import PrayerGathering from "../../images/prayer-gathering.png";
```
Comment (thedomcab) [inline frontend/src/pages/events/index.js:7]:
```suggestion
import { StaticImage } from "gatsby-plugin-image";
```

File: frontend/src/pages/events/index.js [hunk 13]
```diff
@@ -4,11 +4,22 @@ import Layout from "../../components/layout";
...
     <SundayCelebBarEvents />
     <div>Event list section</div>
+    <EventCard
+      title={"Prayer Gathering"}
+      date={"Th, May 25, 2023"}
+      img={../../images/prayer-gathering.png}
+      location={"T Center"}
+      description={`I&apos;m baby migas fam yuccie, big mood freegan affogato everyday
+          carry hashtag four dollar toast truffaut 3 wolf moon beard. Hella
+          live-edge brunch neutral milk hotel pabst.`}
```
Comment (thedomcab) [inline frontend/src/pages/events/index.js:21]:
```suggestion
      title="Prayer Gathering"
      date="Th, May 25, 2023"
     img={<StaticImage
        className="h-[6.15788rem] w-[10.94738rem] flex-shrink-0 mb-0 md:h-60 md:w-[26.66669rem]"
        src="../../images/prayer-gathering.png"
        alt="Background" />}
      location="T Center"
      description="I&apos;m baby migas fam yuccie, big mood freegan affogato everyday
          carry hashtag four dollar toast truffaut 3 wolf moon beard. Hella
          live-edge brunch neutral milk hotel pabst."
```

**Review by thedomcab — COMMENTED** [review]
here is the final suggestions. once done i will approve. test it out on my end and it works.  okay?

**Review by thedomcab — APPROVED** [review]

**Pull request merged**